bentoml/_internal/frameworks/pickle.py

```diff
@@ -0,0 +1,281 @@
+import typing as t
+from typing import TYPE_CHECKING
+
+from simple_di import inject
+from simple_di import Provide
+
+from bentoml import Tag
+from bentoml import Model
+from bentoml import Runner
+from bentoml import SimpleRunner
+from bentoml.exceptions import BentoMLException
+from bentoml.exceptions import MissingDependencyException
+
+from ..types import PathType
+from ..utils import LazyLoader
+from ..models import PKL_EXT
+from ..models import SAVE_NAMESPACE
+from ..utils.pkg import get_pkg_version
+from ..configuration.containers import BentoMLContainer
+
+if TYPE_CHECKING:
+    import numpy as np
+    from .. import ext_typing as ext
+
+    from ..models import ModelStore
+
+import pickle
+
+try:
+    import cloudpickle
+except ImportError:  # pragma: no cover
+    raise MissingDependencyException(
+        "cloudpickle is required in order to use the module `bentoml.pickle`, install cloudpickle with `pip install cloudpickle`."
+    )
+
+MODULE_NAME = "bentoml.pickle"
+
+
+np = LazyLoader("np", globals(), "numpy")  # noqa: F811
+pd = LazyLoader("pd", globals(), "pandas")
+
+
+def _get_model_info(tag: Tag, model_store: "ModelStore") -> t.Tuple["Model", PathType]:
+    model = model_store.get(tag)
+    if model.info.module not in (MODULE_NAME, __name__):
+        raise BentoMLException(
+            f"Model {tag} was saved with module {model.info.module}, cannot load with {MODULE_NAME}."
+        )
+    model_file = model.path_of(f"{SAVE_NAMESPACE}{PKL_EXT}")
+
+    return model, model_file
+
+
+@inject
+def load(
+    tag: t.Union[str, Tag],
+    model_store: "ModelStore" = Provide[BentoMLContainer.model_store],
+) -> t.Any:
+    """
+    Load a model from BentoML local modelstore with given name.
+
+    Args:
+        tag (:code:`Union[str, Tag]`):
+            Tag of a saved model in BentoML local modelstore.
+        model_store (:mod:`~bentoml._internal.models.store.ModelStore`, default to :mod:`BentoMLContainer.model_store`):
+            BentoML modelstore, provided by DI Container.
+
+    Returns:
+        :obj: an instance of :obj: model from BentoML modelstore.
+
+    Examples:
+
+    .. code-block:: python
+
+        import bentoml
+
+        unpickled_model = bentoml.pickle.load('my_model:latest')
+    """  # noqa
+    _, model_file = _get_model_info(tag, model_store)
+    with open(model_file, "rb") as f:
+        # The protocol version used is detected automatically, so we do not
+        # have to specify it.
+        return pickle.load(f)
+
+
+@inject
+def save(
+    name: str,
+    model: t.Any,
```

Comment:
Save should allow user to specify the `method` to call, on the object being pickled.  The default can be `__call__` which works when user saves a function as model.

```suggestion
    obj: t.Any,
    method_name: str = "__call__",
```